Observer/SalesQuoteProductAddAfter.php

```diff
@@ -0,0 +1,183 @@
+<?php
+
+namespace Klaviyo\Reclaim\Observer;
+
+use Klaviyo\Reclaim\Helper\Data;
+
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Catalog\Model\CategoryFactory;
+
+class SalesQuoteProductAddAfter implements ObserverInterface
+{
+    /**
+     * Klaviyo Data Helper
+     * @var Data
+     */
+    protected $_dataHelper;
+
+    /**
+     * Magento Category Factory
+     * @var CategoryFactory
+     */
+    protected $_categoryFactory;
+
+    /**
+     * @param Data $dataHelper
+     * @param CategoryFactory $categoryFactory
+     */
+    public function __construct(
+        Data $dataHelper,
+        CategoryFactory $categoryFactory
+    )
+    {
+        $this->_dataHelper = $dataHelper;
+        $this->_categoryFactory = $categoryFactory;
+    }
+
+    public function execute(Observer $observer)
+    {
+        $quote = $observer->getData('items')[0]->getQuote();
+        $addedItems = $observer->getData('items');
+
+        // Create a list of Simple Product Ids being added as part of a bundle.
+        // We collect the Item name and Qty added to cart, to be sent as AddedItemBundleOptions with the payload.
+        $childrenIds = [];
+
+        foreach ($addedItems as $item){
+            if ($item->getProductType() == 'bundle') {
+                $children = $item->getChildren();
+                foreach ($children as $child){
+                    if (!in_array($child->getId(), $childrenIds)) {
+                        array_push($childrenIds, $child->getId());
+                    }
+                }
+            } elseif (!is_null($item->getParentItem()) && !in_array($item->getId(), $childrenIds)) { //If child comes  before Parent, collect those too
```

Comment:
why is this even performed.  can we just do nothing if the above if condition is the case after all elseifs are satisfied?

Reply:
I guess we could return, but further down in the loop iteration we basically build the cart payload and we don't want to do that for any children Ids, that is why I'm adding it to the array if it has a parentItem associated